DSCResources/MSFT_xADDomainController/MSFT_xADDomainController.psm1

```diff
@@ -164,25 +202,42 @@ function Set-TargetResource
 
 function Test-TargetResource
 {
+    [CmdletBinding()]
     [OutputType([System.Boolean])]
     param
     (
-        [Parameter(Mandatory)]
+        [Parameter(Mandatory = $true)]
+        [System.String]
         [String]$DomainName,
 
-        [Parameter(Mandatory)]
-        [PSCredential]$DomainAdministratorCredential,
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $DomainAdministratorCredential,
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $SafemodeAdministratorPassword,
 
-        [Parameter(Mandatory)]
-        [PSCredential]$SafemodeAdministratorPassword,
+        [Parameter()]
+        [System.String]
+        $DatabasePath,
 
-        [String]$DatabasePath,
+        [Parameter()]
+        [System.String]
+        $LogPath,
 
-        [String]$LogPath,
+        [Parameter()]
+        [System.String]
+        $SysvolPath,
 
-        [String]$SysvolPath,
+        [Parameter()]
+        [System.String]
+        $SiteName,
 
-        [String]$SiteName
+        [Parameter()]
+        [ValidateScript( {Test-Path $_} )]
```

Comment:
For Test-TargetResource, it could forgo the $InstallationMediaPath parameter validation here, especially since the parameter is not used in the function. After the domain has been promoted from media, the user could remove the media. If they did, then subsequent Test-DscConfiguration would fail for that resource due to the validatescript, even though it is a valid scenario.

Reply:
With no Test-Path this is the error. 
![image](https://user-images.githubusercontent.com/8922086/51254526-4a4aef80-196f-11e9-9a44-c3e000f42842.png)

With Test-Path in Set this is the error.
![image](https://user-images.githubusercontent.com/8922086/51254598-66e72780-196f-11e9-9d04-c986fb37bb3a.png)

I like the second better I think. It says it returned Test-Path as $false instead of saying missing parameter.

Reply:
We want the validatescript on Set-TargetResource, but not Get or Test.  The error in the first screenshot looks like the parameter was removed from Get-TargetResource, so when we splat @PsBoundParameters to it, $InstallMediaPath was not a valid parameter.